src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs

```diff
@@ -136,5 +136,88 @@ string DomainMapper(Match match)
                 }
             }
         }
+
+        // These patterns come from real-world customer usages
+
+        [Theory]
+        [InlineData("https://foo.com:443/bar/17/groups/0ad1/providers/Network/public/4e-ip?version=16", "Network/public/4e-ip")]
+        [InlineData("ftp://443/notproviders/17/groups/0ad1/providers/Network/public/4e-ip?version=16", "Network/public/4e-ip")]
+        [InlineData("ftp://443/providersnot/17/groups/0ad1/providers/Network/public/4e-ip?version=16", "Network/public/4e-ip")]
+        public void ExtractResourceUri(string url, string expected)
+        {
+            foreach (RegexOptions options in new[] { RegexOptions.Compiled, RegexOptions.None })
+            {
+                Regex r = new Regex(@"/providers/(.+?)\?", options);
+                Match m = r.Match(url);
+                Assert.True(m.Success);
+                Assert.Equal(2, m.Groups.Count);
+                Assert.Equal(expected, m.Groups[1].Value);
+            }
+        }
+
+        [Theory]
+        [InlineData("IsValidCSharpName", true)]
+        [InlineData("_IsValidCSharpName", true)]
+        [InlineData("__", true)]
+        [InlineData("a\u2169", true)] // \u2169 is in {Nl}
+        [InlineData("\u2169b", true)] // \u2169 is in {Nl}
+        [InlineData("a\u0600", true)] // \u0600 is in {Cf}
+        [InlineData("\u0600b", false)] // \u0600 is in {Cf}
+        [InlineData("a\u0300", true)] // \u0300 is in {Mn}
+        [InlineData("\u0300b", false)] // \u0300 is in {Mn}
+        [InlineData("https://foo.com:443/bar/17/groups/0ad1/providers/Network/public/4e-ip?version=16", false)]
+        [InlineData("david.jones@proseware.com", false)]
```

Comment:
Nit: I see this came from some ancient BCL blog (at least) but I suggest to replace with a foo@contoso.com address that definitely doesn't belong to anyone.

Reply:
> I see this came from some ancient BCL blog

It's actually in the docs:
https://docs.microsoft.com/en-us/dotnet/standard/base-types/how-to-verify-that-strings-are-in-valid-email-format#compile-the-code

I just put all of the docs samples into our tests, basically verbatim.

Reply:
And I just copied it from the above test.

I can change all the emails to `foo.bar@contoso.com` in this file, if you two think it is necessary.

Reply:
Funny enough, `proseware.com` redirects me to `https://www.microsoft.com/en-us/`. So I think it is safe. 😉